### PR TITLE
Initial MAD-X integration

### PR DESCRIPTION
**Pull request by AbbyGi**
This PR adds initial support for running simulations, getting the simulation data file, and creating ophyd classes and objects for MAD-X simulations.

### CODE REVIEW

File: sirepo_bluesky/sirepo_ophyd.py [hunk 1]
```diff
@@ -92,8 +95,10 @@ def __init__(
         sim_type = self.connection.data["simulationType"]
         allowed_sim_types = ("srw", "shadow")
         if sim_type not in allowed_sim_types:
```
Comment (jklynch) [inline sirepo_bluesky/sirepo_ophyd.py:97]: should 'madx' be an allowed simulation type?

**Review by jklynch — CHANGES_REQUESTED** [review]
is it possible to add a madx test?

File: sirepo_bluesky/sirepo_ophyd.py (hunk 1 above)
Comment (mrakitin) [reply sirepo_bluesky/sirepo_ophyd.py:97]: Good point, thanks for catching this, @jklynch!

**Review by mrakitin — COMMENTED** [review]
Thanks, @AbbyGi! I have a suggestion to generalize the way we receive the N-th file from Sirepo. Also, it would be great to address @jklynch's comment. Thanks!

**Review by jennmald — APPROVED** [review]

File: sirepo_bluesky/sirepo_bluesky.py [hunk 2]
```diff
@@ -112,18 +112,15 @@ def find_optic_id_by_name(self, optic_name):
...
             Notes
             -----
             Call auth() and run_simulation() before this.
         """
         assert hasattr(self, 'cookies'), 'call auth() before get_datafile()'
-        if self.sim_type in ["srw", "shadow"]:
-            url = 'download-data-file/{}/{}/{}/-1'.format(self.sim_type, self.sim_id, self.data['report'])
-        elif self.sim_type in ["madx"]:
-            url = 'download-data-file/{}/{}/{}/0'.format(self.sim_type, self.sim_id, self.data['report'])
+        url = f"download-data-file/{self.sim_type}/{self.sim_id}/{self.data['report']}/{file_index}"
```
Comment (mrakitin) [inline sirepo_bluesky/sirepo_bluesky.py:123]: Probably we need to address the case of `file_index=0` in the relevant code that uses this method. Could you please have a look?

**Review by mrakitin — COMMENTED** [review]
Good addition. It would be really helpful to have a simple test for the existing simulation from https://github.com/NSLS-II/sirepo-bluesky/tree/master/sirepo_bluesky/tests/SIREPO_SRDB_ROOT/user/testuser/madx (or feel free to add your .json with the predefined number, e.g. `00000000`). I left a comment about the use of the `file_index` kwarg.

**Note** to other reviewers: I would like to merge this PR after we release #70 and #72.

**Comment by mrakitin** [comment]
Converted it to "Draft" to prevent accidental merging.

File: sirepo_bluesky/sirepo_bluesky.py [hunk 3]
```diff
@@ -120,7 +120,10 @@ def get_datafile(self):
             Call auth() and run_simulation() before this.
         """
         assert hasattr(self, 'cookies'), 'call auth() before get_datafile()'
-        url = 'download-data-file/{}/{}/{}/-1'.format(self.sim_type, self.sim_id, self.data['report'])
+        if self.sim_type in ["srw", "shadow"]:
+            url = 'download-data-file/{}/{}/{}/-1'.format(self.sim_type, self.sim_id, self.data['report'])
+        elif self.sim_type in ["madx"]:
+            url = 'download-data-file/{}/{}/{}/0'.format(self.sim_type, self.sim_id, self.data['report'])
```
[the remark this answers is not in the document]
Comment (mrakitin) [reply sirepo_bluesky/sirepo_bluesky.py]: I think it's resolved in recent commits.

**Review by mrakitin — COMMENTED** [review]
Looks good to me, here are a couple of clean-up suggestions.

**Review by jklynch — APPROVED** [review]
Nice tests, @AbbyGi.

**Review by mrakitin — APPROVED** [review]
Great team work on that!

**Pull request merged**